dags/veda_data_pipeline/veda_dataset_pipeline.py

```diff
@@ -83,19 +64,38 @@ def build_stac_task(payload):
 }
 
 with DAG("veda_dataset_pipeline", params=template_dag_run_conf, **dag_args) as dag:
-    # ECS dependency variable
+    start = EmptyOperator(task_id="start")
+    end = EmptyOperator(task_id="end")
 
-    start = EmptyOperator(task_id="start", dag=dag)
-    end = EmptyOperator(task_id="end", dag=dag)
 
-    collection_grp = collection_task_group()
-    discover = discover_from_s3_task.expand(event=extract_discovery_items())
-    discover.set_upstream(collection_grp)  # do not discover until collection exists
-    get_files = get_dataset_files_to_process(payload=discover)
+    @task()
+    def mutate_payload(**kwargs):
+        ti = kwargs.get("ti")
+        payload = ti.dag_run.conf.copy()
+        payloads = list()
+        if assets := payload.get("assets"):
+            # remove thumbnail asset if provided in collection config
+            if "thumbnail" in assets.keys():
+                assets.pop("thumbnail")
+            # if thumbnail was only asset, delete assets
+            if not assets:
+                payload.pop("assets")
+            # finally put the mutated assets back in the payload
+            else:
+                payload["assets"] = assets
+        for item in payload.get("discovery_items"):
+            payloads.append({
+                **payload,
+                **item
+            }
+            )
```

Comment:
This adds a lot of additional unknown properties to the payload, which will cause downstream issues (especially as we add enforced schemas and `Param` definitions)